covertreemap/percent_coverage.go

```diff
@@ -0,0 +1,25 @@
+package covertreemap
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/nikolaydubina/treemap"
+)
+
+func AddCoveragePercentageToName(tree *treemap.Tree) {
+	for path, node := range tree.Nodes {
+		parts := strings.Split(node.Path, "/")
+		if len(parts) == 0 {
+			continue
+		}
+
+		tree.Nodes[path] = treemap.Node{
+			Path:    node.Path,
+			Name:    parts[len(parts)-1] + " " + strconv.FormatFloat(node.Heat*100, 'f', 1, 64) + "%",
```

Comment:
holdon, why we throwing away full path except leaf node? please keep all.

it actually makes sense to allow users to keep only leaf node, but that has to be separate option. feel free to raise another PR!
```suggestion
			Name:    path + " " + strconv.FormatFloat(node.Heat*100, 'f', 1, 64) + "%",
```

Reply:
but in SetNamesFromPath this is exactly how the name is set, and I only add the percent output after ImputeHeat and CollapseLongPaths are executed, am I wrong?

and of course I have to use strings.Builder for concatenation

```
// SetNamesFromPaths will update each node to its path leaf as name.
func SetNamesFromPaths(t *Tree) {
	if t == nil {
		return
	}
	for path, node := range t.Nodes {
		parts := strings.Split(node.Path, "/")
		if len(parts) == 0 {
			continue
		}

		t.Nodes[path] = Node{
			Path:    node.Path,
			Name:    parts[len(parts)-1] ,
			Size:    node.Size,
			Heat:    node.Heat,
			HasHeat: node.HasHeat,
		}
	}
}
```

Reply:
holdon. let me test more examples with path folding...

Reply:
ok, the output is different. just run this on hugo repo.

master
![out](https://github.com/user-attachments/assets/66f0fa60-657b-44ae-95be-0c607813035b)

your PR
![out2](https://github.com/user-attachments/assets/89081470-d97b-4ee4-83dc-7cf499099af7)

notice this

before
<img width="343" alt="image" src="https://github.com/user-attachments/assets/6b6f3541-b84f-4201-873c-b8c194d5f31f">


now, paths missing
<img width="339" alt="image" src="https://github.com/user-attachments/assets/295d5e26-f64d-4163-9427-52abcee9f2d9">

Reply:
so, now I've corrected this, the fact is that I didn't see that CollapseLongPaths was converting the Name for all Nodes except Root